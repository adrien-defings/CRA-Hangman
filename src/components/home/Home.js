import React from 'react'
import PropTypes from 'prop-types'

import "./home.css"

const Home = ({ onClick }) => {
  return (
    <div id="home">
      {/* classic home display */}
      <h4>Welcome <span role="img" aria-label="hangman logo">🙋‍♂️</span></h4>
      <p className="lead">Please choose one of this following navigationMode :</p>
      <button className="btn btn-primary" onClick={() => onClick("singlePlayer")}>Single Player</button>
      <button className="btn btn-primary" onClick={() => onClick("multiplayer")}>Multiplayer</button>
    </div>
  )
}

Home.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default Home