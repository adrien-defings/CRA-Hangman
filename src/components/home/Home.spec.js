import React from 'react'
import { shallow } from 'enzyme'

import Home from './Home'

// WITH JEST + ENZYME
//  it('should trigger its `onClick` on singlePlayer Mode', () => {
//    const onClick = jest.fn()
//    const wrapper = shallow(
//      <Home onClick={onClick} />
//    )

//    wrapper.find('button').at(0).simulate('click')
//    expect(onClick).toHaveBeenCalledWith("singlePlayer")
//  })

import { expect } from 'chai'
import sinon from 'sinon'

// With Chai-Enzyme + sinon
describe('<Home />', () => {
  let onClick
  let wrapper

  // Render the component in the beforeEach
  // => Each test can get a new shallow rendered wrapper
  beforeEach(() => {
    onClick = sinon.spy()
    wrapper = shallow(
      <Home onClick={onClick} />
    )
    expect(wrapper).to.matchSnapshot()
  })

  it('should trigger its `onClick` on singlePlayer Mode', () => {
    wrapper.find('button').at(0).simulate('click')
    expect(onClick).to.have.been.calledWith("singlePlayer")
  })

  it('should trigger its `onClick` on multiplayer Mode', () => {
    wrapper.find('button').at(1).simulate('click')
    expect(onClick).to.have.been.calledWith("multiplayer")
  })
})