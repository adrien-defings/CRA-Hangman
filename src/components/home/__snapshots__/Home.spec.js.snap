// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should trigger its \`onClick\` on multiplayer Mode 1`] = `
<div
  id="home"
>
  <h4>
    Welcome 
    <span
      aria-label="hangman logo"
      role="img"
    >
      🙋‍♂️
    </span>
  </h4>
  <p
    className="lead"
  >
    Please choose one of this following navigationMode :
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Single Player
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Multiplayer
  </button>
</div>
`;

exports[`<Home /> should trigger its \`onClick\` on singlePlayer Mode 1`] = `
<div
  id="home"
>
  <h4>
    Welcome 
    <span
      aria-label="hangman logo"
      role="img"
    >
      🙋‍♂️
    </span>
  </h4>
  <p
    className="lead"
  >
    Please choose one of this following navigationMode :
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Single Player
  </button>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Multiplayer
  </button>
</div>
`;
